### PR TITLE
x86: update x86inc

### PR DESCRIPTION
**Pull request by CoffeeFlux**
This should fix the warning introduced with nasm 2.15

### CODE REVIEW

**Comment by TheOneric** [comment]
It seems like we'll also need to update our `ASFLAGS` along with this update; looking at the macros we set:
 - `HAVE_ALIGNED_STACK` was removed, instead `STACK_ALIGNMENT` can be used where necessary I guess
 - `PIC` is now handled differently *(can we just always set it now that it is force-disabled for win32?)*
- `HAVE_CPUNOP` was removed

**Comment by CoffeeFlux** [comment]
`HAVE_ALIGNED_STACK` checks have been replaced with `STACK_ALIGNMENT == 16`.

`PIC` is no longer defined in the configure scripts, since it seems like the existing behavior will just be preserved with the new x86inc internal logic. If we want to always set it I can do that instead, but I don't have context on how this impacts 32-bit linux so it's not my call to make.

`HAVE_CUPNOP` is no longer defined.

**Comment by TheOneric** [comment]
> PIC is no longer defined in the configure scripts, since it seems like the existing behavior will just be preserved with the new x86inc internal logic. If we want to always set it I can do that instead, but I don't have context on how this impacts 32-bit linux

Afaik it should be fine on all supported i386 Linux kernels and to my knowledge non-PIC code being included can break ASLR and potentially cause issues in shared libs. Not being PIC was also the reason ASM got [disabled in Debian's i386-builds of libass](https://salsa.debian.org/multimedia-team/libass/-/commit/d9eae1f0aefacf9b8c838cb5108dcc100f336e0b). 

However I'm not completely sure that always enabling PIC won't cause problems on some other non-Windows system and I realised it is possible for packagers to pass `ASFLAGS=-DPIC` to `configure` to enable it if it is preferred.
Given that this better matches the current behaviour, it's probably safer to leave the decision about PIC on i386 to x86inc's defaults and user `ASFLAGS` instead of always enabling it, as I previously hinted at. So I think this should be fine.

---

By changing `configure.ac` this pr now conflicts with #452. As #452 basically touches almost _everything_ in `configure.ac`, I'd prefer to merge #452 _before_ this one, as merging it the other way around would result in all of #452's commits needing to be adjusted, meaning new errors could slip in and I'd have to reverify it on all 6 systems.
Merging #452 first on the other hand only requires a few lines of this patch to be adjusted.

File: configure.ac [hunk 1]
```diff
@@ -207,15 +207,13 @@ AS_IF([test x$enable_asm != xno], [
         ], [
             AS_CASE([$host],
                 [*darwin*], [
-                    ASFLAGS="$ASFLAGS -f macho$BITTYPE -DPREFIX -DHAVE_ALIGNED_STACK=1" ],
+                    ASFLAGS="$ASFLAGS -f macho$BITTYPE -DPREFIX" ],
                 [*linux*|*dragonfly*|*bsd*|*solaris*|*haiku*], [
-                    ASFLAGS="$ASFLAGS -f elf$BITTYPE -DHAVE_ALIGNED_STACK=1" ],
+                    ASFLAGS="$ASFLAGS -f elf$BITTYPE" ],
```
Comment (TheOneric) [inline configure.ac:212]: Won't this change stack alignment on 32-bit MacOS, Linux, Solaris and Haiku? Afaict x86inc defaults to what would previously have been `HAVE_ALIGNED_STACK=0` on i386-systems.
 *(Dragonfly and other BSDs shouldn't have 16Byte alignment on 32-bit anyway as demonstrated by #451 and fixed as part of #452)*

**Comment by rcombs** [comment]
The reason lots of code doesn't support PIC on i386 (including ffmpeg, which debian seems to have no issue continuing to package with assembly enabled) is that there's no IP-relative addressing mode, so doing data loads ends up being really complex and expensive (it requires use of at least one extra register, and i386 is already pretty starved for those). I don't see much reason to try to port stuff to PIC-i386 at this point tbh; it's not like there are a whole lot of people clamoring for it when just about everybody not on windows has moved to x86_64.

**Comment by CoffeeFlux** [comment]
I'm sympathetic to your reasons for merging #452 first, particularly given I've had the meson PR sitting on the vine for a couple years now, but that's a rather large PR making mostly cosmetic changes and this one fixes an immediate issue. But well, not my call either way.

Will check on the stack alignment.

File: libass/x86/rasterizer.asm [hunk 2]
```diff
@@ -488,7 +488,7 @@ endstruc
 ;------------------------------------------------------------------------------
 
 %macro GET_RES_ADDR 1
-%if mmsize <= 16 && HAVE_ALIGNED_STACK
+%if mmsize <= 16 && STACK_ALIGNMENT == 16
```
Comment (MrSmile) [inline libass/x86/rasterizer.asm:491]: To correctly support higher stack alignments that should be
```asm
%if mmsize <= STACK_ALIGNMENT
```

Comment (CoffeeFlux) [reply libass/x86/rasterizer.asm:491]: Right, sorry. Fixed.

**Review by rcombs — APPROVED** [review]

**Comment by TheOneric** [comment]
With #452 merged, and relying on `x86inc.asm` alignment defaults, I think the configure.ac-part should now look like this:
```diff
diff --git a/configure.ac b/configure.ac
index 2e5cae6..6cb4948 100644
--- a/configure.ac
+++ b/configure.ac
@@ -196,7 +196,7 @@ AS_IF([test "x$enable_asm" != xno], [
             X64=true
             BITS=64
             BITTYPE=x32
-            ASFLAGS="$ASFLAGS -DARCH_X86_64=1 -DPIC"
+            ASFLAGS="$ASFLAGS -DARCH_X86_64=1"
         ],
         [x86_64-*|amd64-*], [
             AS=nasm
@@ -204,7 +204,7 @@ AS_IF([test "x$enable_asm" != xno], [
             X64=true
             BITS=64
             BITTYPE=64
-            ASFLAGS="$ASFLAGS -DARCH_X86_64=1 -DPIC"
+            ASFLAGS="$ASFLAGS -DARCH_X86_64=1"
         ],
         [ # default
             INTEL=false
@@ -219,25 +219,18 @@ AS_IF([test "x$enable_asm" != xno], [
         ], [
             AS_CASE([$host],
                 [*darwin*], [
-                    ASFLAGS="$ASFLAGS -f macho$BITTYPE -DPREFIX -DHAVE_ALIGNED_STACK=1"
+                    ASFLAGS="$ASFLAGS -f macho$BITTYPE -DPREFIX -DSTACK_ALIGNMENT=16"
                 ],
                 [*linux*|*solaris*|*haiku*], [
-                    ASFLAGS="$ASFLAGS -f elf$BITTYPE -DHAVE_ALIGNED_STACK=1"
+                    ASFLAGS="$ASFLAGS -f elf$BITTYPE -DSTACK_ALIGNMENT=16"
                 ],
                 [*dragonfly*|*bsd*], [
                     ASFLAGS="$ASFLAGS -f elf$BITTYPE"
-                    AS_IF([test "x$BITS" = x64], [
-                        ASFLAGS="$ASFLAGS -DHAVE_ALIGNED_STACK=1"
-                    ], [
-                        ASFLAGS="$ASFLAGS -DHAVE_ALIGNED_STACK=0"
-                    ])
                 ],
                 [*cygwin*|*mingw*], [
                     ASFLAGS="$ASFLAGS -f win$BITTYPE"
-                    AS_IF([test "x$BITS" = x64], [
-                        ASFLAGS="$ASFLAGS -DHAVE_ALIGNED_STACK=1"
-                    ], [
-                        ASFLAGS="$ASFLAGS -DHAVE_ALIGNED_STACK=0 -DPREFIX"
+                    AS_IF([test "x$BITS" = x32], [
+                        ASFLAGS="$ASFLAGS -DPREFIX"
                     ])
                 ],
                 [ # default
@@ -251,7 +244,7 @@ AS_IF([test "x$enable_asm" != xno], [
                     ))
                 ]
             )
-            ASFLAGS="$ASFLAGS -DHAVE_CPUNOP=0 -Dprivate_prefix=ass"
+            ASFLAGS="$ASFLAGS -Dprivate_prefix=ass"
             AC_MSG_CHECKING([if $AS supports vpmovzxwd])
             echo "vpmovzxwd ymm0, xmm0" > conftest.asm
             AS_IF([$AS conftest.asm $ASFLAGS -o conftest.o >conftest.log 2>&1], [

```

**Review by TheOneric — APPROVED** [review]

**Pull request merged**